### PR TITLE
use trace instead of simulate to get txn receipt

### PR DESCRIPTION
**Pull request by apoorvsadana**
1. Uses trace instead of simulation for transaction receipt. Trace has re execution support.
2. Events and messages don't work in the case of pending receipts (didn't work before either)
3. 2 can be fixes right now but its easier to do when Massa labs implements pending blocks.

### CODE REVIEW

File: crates/client/rpc/src/lib.rs [hunk 1]
```diff
@@ -1599,16 +1601,20 @@ where
 
         // TODO
         // Is any better way to get execution resources of processed tx?
-        let skip_validate = true;
-        // let parent_block_hash = Felt252Wrapper::from().into();
-        let parent_block_hash = self
-            .substrate_block_hash_from_starknet_block(BlockId::Hash(block_header.parent_block_hash.into()))
+        let parent_substrate_block_hash = self
+            .substrate_block_hash_from_starknet_block(BlockId::Hash(starknet_block.header().parent_block_hash.into()))
```
Comment (EvolveArt) [inline crates/client/rpc/src/lib.rs:1605]:
```suggestion
            .substrate_block_hash_from_starknet_block(BlockId::Hash(block_header.parent_block_hash.into()))
```

File: crates/client/rpc/src/lib.rs [hunk 2]
```diff
@@ -1599,16 +1601,20 @@ where
...
-        let simulation = self.simulate_tx(parent_block_hash, transaction.clone(), skip_validate, fee_disabled)?;
-        let execution_resources = actual_resources_to_execution_resources(simulation.actual_resources);
+        let execution_info = self.get_transaction_execution_info(
+            parent_substrate_block_hash,
+            starknet_block.transactions(),
+            chain_id,
+            transaction_hash,
+        )?;
+
+        let execution_resources = actual_resources_to_execution_resources(execution_info.actual_resources.clone());
```
Comment (EvolveArt) [inline crates/client/rpc/src/lib.rs:1617]: need to clone?

File: starknet-rpc-test/Cargo.toml [hunk 3]
```diff
@@ -120,9 +120,9 @@ path = "estimate_message_fee.rs"
 name = "starknet_simulate_transaction"
 path = "simulate_transaction.rs"
 
-[[test]]
-name = "starknet_get_state_update"
-path = "get_state_update.rs"
+#[[test]]
```
Comment (EvolveArt) [inline starknet-rpc-test/Cargo.toml:123]: why ?

File: crates/client/rpc/src/lib.rs (hunk 1 above)
Comment (apoorvsadana) [reply crates/client/rpc/src/lib.rs]: fixed

File: crates/client/rpc/src/lib.rs (hunk 2 above)
Comment (apoorvsadana) [reply crates/client/rpc/src/lib.rs]: apparently not, fixed

File: starknet-rpc-test/Cargo.toml (hunk 3 above)
Comment (apoorvsadana) [reply starknet-rpc-test/Cargo.toml]: pushed it by mistake, fixed

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**